src/ParseObject.js

```diff
@@ -179,10 +179,10 @@ class ParseObject {
    * Returns a local or server Id used uniquely identify this object
    */
   _getId(): string {
-    if (typeof this.id === 'string') {
+    if (this.id && typeof this.id === 'string') {
```

Comment:
Can you revert this change? I don't think this is needed. You can write integration test to verify.

Reply:
@dplewis as you wish I've written a unit test but it's maybe useless to add this check

Reply:
An integration tests is different from unit tests. Integration tests run against a parse server instance. Add your tests [here](https://github.com/parse-community/Parse-SDK-JS/blob/master/integration/test/ParseObjectTest.js).